backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java

```diff
@@ -470,20 +475,26 @@ public List<I_C_InvoiceCandidate_InOutLine> retrieveICIOLAssociationsFor(@NonNul
 
 	public boolean isCompletedOrClosedInvoice(@NonNull final InOutId inOutId)
 	{
-		final IQuery<I_C_InvoiceCandidate_InOutLine> candidateInOutLine = queryBL.createQueryBuilder(I_M_InOutLine.class)
-				.addEqualsFilter(I_M_InOutLine.COLUMNNAME_M_InOut_ID, inOutId)
-				.addOnlyActiveRecordsFilter()
-				.andCollectChildren(I_C_InvoiceCandidate_InOutLine.COLUMNNAME_M_InOutLine_ID, I_C_InvoiceCandidate_InOutLine.class)
-				.create();
+		final IQueryBuilder<I_C_Invoice_Line_Alloc> invoiceLineAllocQueryBuilder = retrieveInvoiceLineAllocQueryBuilder(inOutId);
 
-		return queryBL.createQueryBuilder(I_C_Invoice_Line_Alloc.class)
-				.addOnlyActiveRecordsFilter()
-				.addInSubQueryFilter(I_C_Invoice_Line_Alloc.COLUMNNAME_C_Invoice_Candidate_ID, I_C_InvoiceCandidate_InOutLine.COLUMNNAME_C_Invoice_Candidate_ID, candidateInOutLine)
-				.andCollect(I_C_InvoiceLine.COLUMNNAME_C_InvoiceLine_ID, I_C_InvoiceLine.class)
-				.andCollect(I_C_Invoice.COLUMNNAME_C_Invoice_ID, I_C_Invoice.class)
-				.addInArrayOrAllFilter(I_C_Invoice.COLUMNNAME_DocStatus, IDocument.STATUS_Closed, IDocument.STATUS_Completed) // DocStatus in ('CO', 'CL')
-				.create()
-				.anyMatch();
+		if (isAggregationWithAttributeAggregatePerInOut(invoiceLineAllocQueryBuilder))
+		{
+			return queryBL.createQueryBuilder(I_C_Invoice.class)
+					.addOnlyActiveRecordsFilter()
+					.addEqualsFilter(COLUMNNAME_M_InOut_ID, inOutId)
+					.addInArrayOrAllFilter(I_C_Invoice.COLUMNNAME_DocStatus, IDocument.STATUS_Closed, IDocument.STATUS_Completed) // DocStatus in ('CO', 'CL')
+					.create()
+					.anyMatch();
+		}
+		else
+		{
+			return invoiceLineAllocQueryBuilder
```

Comment:
doesn't ` invoiceLineAllocQueryBuilder` at this point have all the filters and "andCollect"s that were added in the above if-clause?

Reply:
You're right about `M_MatchInv`, changed the implementation to use it